### ISSUE TITLE
CA1802 could also flag some instance fields

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#2772]
**Version Used**: 
3.3.0-beta3-19409-05+ed92d532473db83c1db47b313ee1c1bd7520aa08

**Steps to Reproduce**:
```C#
using System;

class Program
{
    static void Main()
    {
        foreach (string s in new Splitter().Split("hello,world"))
            Console.WriteLine(s);
    }
}

class Splitter
{
    private readonly char _separator = ',';

    public string[] Split(string s) => s.Split(_separator);
}
```

**Expected Behavior**:
CA1802 would suggest refactoring `_separator` to be `const`.

**Actual Behavior**:
CA1802 ignores `_separator` because it's an instance rather than static field.

**Comment by mavasani** [comment]
Analyzer seems to match the legacy FxCop behavior, which only flagged static fields: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1802-use-literals-where-appropriate?view=vs-2019.

We can add a new [analyzer configuration option](https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md) to allow configuring if instance fields should be flagged or not.

**Comment by JohanLarsson** [comment]
Tangent: static locals would be nice for this kind of thing. Keep related things close and with minimal scope.
